hist/hist/test/test_TH1_SaveAs.cxx

```diff
@@ -0,0 +1,224 @@
+         "2	3	12.3	2.1",
+         "3	4	9.5	3.5",
+         "4	5	7.3	2.7",
+         "5	6	15.2	4.7"
+      };
+      std::string line;
+      while (std::getline(infile, line)) {
+         idx++;
+         if (idx>N) {
+            infile.close();
+            return 0;
+         }
+         if (line != ref[idx-1]) {
+            infile.close();
+            return 0;
+         }
+      }
+      infile.close();
+      return 1;
+   }
+
+   bool IsGood_txt() {
+      std::sprintf(filename,"%s%s",fnam.Data(),"txt");
+      std::ifstream infile(filename, std::ios::in);
+      if (!infile) {
+         return 0;
+      }
+      Int_t idx = 0;
+      TString ref[N] = {
+         "-1 0 5.2 2.1",
+         "0 1 0 0",
+         "1 2 10.8 2.5",
+         "2 3 12.3 2.1",
+         "3 4 9.5 3.5",
+         "4 5 7.3 2.7",
+         "5 6 15.2 4.7"
+      };
+      std::string line;
+      while (std::getline(infile, line)) {
+         idx++;
+         if (idx>N) {
+            infile.close();
+            return 0;
+         }
+         if (line != ref[idx-1]) {
+            infile.close();
+            return 0;
+         }
+      }
+      infile.close();
+      return 1;
+   }
+
+   bool IsGood_C() {
+      std::sprintf(filename,"%s%s",fnam.Data(),"C");
+      std::ifstream infile(filename, std::ios::in);
+      if (!infile) {
+         return 0;
+      }
+      const Int_t NC = 29;  // lines in C file (excl. empty and commented out lines)
+      Int_t idx = 0;
+      TString ref[NC] = {
+         "{",
+         "   TH1D *h__1 = new TH1D(\"h__1\",\"h_title\",5,0,5);",
+         "   h__1->SetBinContent(0,5.2);",
+         "   h__1->SetBinContent(2,10.8);",
+         "   h__1->SetBinContent(3,12.3);",
+         "   h__1->SetBinContent(4,9.5);",
+         "   h__1->SetBinContent(5,7.3);",
+         "   h__1->SetBinContent(6,15.2);",
+         "   h__1->SetBinError(0,2.1);",
+         "   h__1->SetBinError(2,2.5);",
+         "   h__1->SetBinError(3,2.1);",
+         "   h__1->SetBinError(4,3.5);",
+         "   h__1->SetBinError(5,2.7);",
+         "   h__1->SetBinError(6,4.7);",
+         "   h__1->SetEntries(7);",
+         "   Int_t ci;      // for color index setting",
+         "   TColor *color; // for color definition with alpha",
+         "   ci = TColor::GetColor(\"#000099\");",
+         "   h__1->SetLineColor(ci);",
+         "   h__1->GetXaxis()->SetLabelFont(42);",
+         "   h__1->GetXaxis()->SetTitleOffset(1);",
+         "   h__1->GetXaxis()->SetTitleFont(42);",
+         "   h__1->GetYaxis()->SetLabelFont(42);",
+         "   h__1->GetYaxis()->SetTitleFont(42);",
+         "   h__1->GetZaxis()->SetLabelFont(42);",
+         "   h__1->GetZaxis()->SetTitleOffset(1);",
+         "   h__1->GetZaxis()->SetTitleFont(42);",
+         "   h__1->Draw(\"\");",
+         "}"
+      };
+      std::string line;
+      while (std::getline(infile, line)) {
+         if ( line!="{" && line!="}" && ( line.rfind("//",0)==0 || line.length()<6 ) ) {
```

Comment:
What is this if condition matching exactly?

Reply:
The generated .C file (created elswhere in ROOT, so I don't control it) contains lines like these at the beginning:
```
{
//========= Macro generated from object: h/h_title
//========= by ROOT version6.31/01
   
```
Where the last line seems, but is not, empty; it has 3 spaces (from indentation). So the "if" is meant to ignore similar comment and empty lines. I did not compare against these exact lines in order to leave a bit of room for possible future changes in exporting .C code, but bigger changes may break this test.

Reply:
Understood, thanks!